### ISSUE TITLE
Text and Attention Masking

### ISSUE THREAD

**Issue by arjun-gangwar** [SWivid/F5-TTS#1184]
### Checks

- [x] This template is only for research question, not usage problems, feature requests or bug reports.
- [x] I have thoroughly reviewed the project documentation and read the related paper(s).
- [x] I have searched for existing issues, including closed ones, no similar questions.
- [x] I am using English to submit this issue to facilitate community communication.

### Question details

Hi,

I have noticed that in some of the configs, text masking and attention masking is set to True while in other it is False. I would imagine you would always want to set both to True. I was just wondering what is the reason for this and what works best? Thanks!

**Comment by SWivid** [comment]
https://github.com/SWivid/F5-TTS/releases/tag/1.0.0
https://swivid.github.io/F5-TTS_updates/

**Comment by arjun-gangwar** [comment]
Thanks for the quick response! 

So, it is better to do text masking in both filler tokens and padding tokens. 

What about attention mask being set to False? I would imagine that we won't want the model to look at the padding tokens during self attention.

**Comment by SWivid** [comment]
Recommand to use flash_attn and enable mask, set true cuz native pytorch with mask is much slower (as you can see in the dataset related code, data samples are sorted in lenght to form batches with similiar length samples inside)

**Comment by arjun-gangwar** [comment]
> Recommand to use flash_attn and enable mask, set true cuz native pytorch with mask is much slower (as you can see in the dataset related code, data samples are sorted in lenght to form batches with similiar length samples inside)

Okay.. so dynamic batch sampling would reduce the amount of padding in each batch. So, we can still use native torch attention without masking to make it faster but it is better to use flash_attn with mask. Makes sense.. thanks a lot!
